src/index.ml

```diff
@@ -426,7 +468,13 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
       |> EntrySet.elements
     in
     merge_with log t tmp;
+    let tmp_fan_path = t.root // "tmp" // "fan" in
+    let fan_path = t.root // "fan" in
+    let tmp_fan = IO.v ~fresh:true ~readonly:false ~generation tmp_fan_path in
+    Fan.export t.fan_out tmp_fan;
+    let fan = IO.v ~fresh:true ~readonly:false ~generation fan_path in
     IO.rename ~src:tmp ~dst:t.index;
+    IO.rename ~src:tmp_fan ~dst:fan;
```

Comment:
beware: this is not atomic anymore. What happens it the program crashes between line 476 and 477?